app/utils/rentry/wrapper.py

```diff
@@ -108,100 +140,150 @@ def new(self, text: str) -> Any:
+        rentry_link = self.link_input[0]  # Get the link from user input
 
         if not self.is_valid_rentry_link(rentry_link):
             logger.warning("Invalid Rentry link. Please enter a valid Rentry link.")
-            # Show warning message box
             show_warning(
                 title="Invalid Link",
                 text="Invalid Rentry link. Please enter a valid Rentry link.",
             )
-            return
+            return self.input_dialog()  # Re-initialize the UI for new input
 
         try:
-            if rentry_link.endswith("/raw"):
-                raw_url = rentry_link
-            else:
-                raw_url = f"{rentry_link}/raw"
-
-            response = requests.get(raw_url)
+            # Determine the raw URL based on the provided link
+            raw_url = (
+                rentry_link if rentry_link.endswith("/raw") else f"{rentry_link}/raw"
+            )
+            response = requests.get(raw_url)  # Fetch the content from the raw URL
 
             if response.status_code == 200:
                 # Decode the content using UTF-8
                 page_content = response.content.decode("utf-8")
 
                 # Define regex pattern for both variations of 'packageid' and 'packageId'
                 pattern = r"(?i){packageid:\s*([\w.]+)\}|packageid:\s*([\w.]+)"
-                matches = re.findall(pattern, page_content)
+                matches = re.findall(
+                    pattern, page_content
+                )  # Find all matches in the content
                 self.package_ids = [
                     match[0] if match[0] else match[1]
                     for match in matches
                     if match[0] or match[1]
                 ]
                 logger.info("Parsed package_ids successfully.")
             else:
-                logger.warning(
-                    f"Rentry returned status code: {response.status_code}. Reason: {response.reason}"
-                )
-                InformationBox(
-                    title="Failed to fetch Rentry Content",
-                    text=f"Rentry returned status code: {response.status_code}",
-                    information="RimSort failed to fetch the content from the provided Rentry link. This may be due to an invalid link, your internet connection, or Rentry.co being down. It may also be the result of a captcha. Please try again later.",
-                    details=response.reason,
-                    icon=QMessageBox.Icon.Warning,
-                ).exec()
+                RentryError().show_info(response)  # Handle non-200 responses
 
         except Exception as e:
-            logger.error(
-                f"An error occurred while fetching rentry.co content: {str(e)}"
-            )
-            show_fatal_error(
-                title="Error",
-                text=f"An error occurred: {str(e)}",
-            )
+            RentryError().show_error_warning(e)  # Handle exceptions during the process
```

Comment:
This should handle normal exceptions. Add except for network exceptions like ```RequestException``` and then create a method to handle the network exceptions (show warning etc. whatever it needs to do)

Reply:
done